File: 2026-01-24-fibonacci-sequence/java/Solution.java
public class Fibonacci {

    public static long getNth(int n) {
        if (n < 0) {
            throw new IllegalArgumentException("n must be non-negative");
        }
        
        if (n == 0) return 0;
        if (n == 1) return 1;

        long prev = 0;
        long curr = 1;

        for (int i = 2; i <= n; i++) {
            long next = prev + curr;
            prev = curr;
            curr = next;
        }

        return curr;
    }
}

File: 2026-01-24-fibonacci-sequence/python/solution.py
def get_nth(n):
    if n < 0:
        raise ValueError("n must be non-negative")
    
    if n == 0:
        return 0
    if n == 1:
        return 1

    prev = 0
    curr = 1

    for i in range(2, n + 1):
        prev, curr = curr, prev + curr

    return curr
